Thorium-Sim/thorium-nova
# Ship System Editor UI

## Description

Add the ability to create ship system instances and edit their properties from the plugin config UI. Impulse engines only for now. 



## Related Issue








Closes #251

## How do you know the changes work correctly?






Manual Testing:
- Navigate to the Ship Systems plugin editor
- Create a new ship system from the dropdown (only generic and impulse engines for now)
- Change some properties and reload the page. They should have persisted.

## Screenshots (if appropriate):

<img width="1792" alt="Screen Shot 2022-04-21 at 9 40 12 PM" src="https://user-images.githubusercontent.com/6558157/164579735-d2833573-8363-47bf-b698-c34f96d67c9b.png">

## Types of changes



      change)

## Checklist:
